domain/src/Aggregates/Nft/Reactors/NftReactor.php

```diff
@@ -25,23 +25,11 @@ public function handleNftDisposedOf(NftDisposedOf $event, Message $message): voi
         $taxYearId = TaxYearId::fromTaxYear($taxYear);
         $taxYearAggregate = $this->taxYearRepository->get($taxYearId);
 
-        if ($event->proceeds->isGreaterThan($event->costBasis)) {
-            $taxYearAggregate->recordCapitalGain(new RecordCapitalGain(
-                taxYear: $taxYear,
-                date: $event->date,
-                amount: $event->proceeds->minus($event->costBasis),
-                costBasis: $event->costBasis,
-                proceeds: $event->proceeds,
-            ));
-        } else {
-            $taxYearAggregate->recordCapitalLoss(new RecordCapitalLoss(
-                taxYear: $taxYear,
-                date: $event->date,
-                amount: $event->costBasis->minus($event->proceeds),
-                costBasis: $event->costBasis,
-                proceeds: $event->proceeds,
-            ));
-        }
+        $taxYearAggregate->updateCapitalGain(new UpdateCapitalGain(
+            taxYear: $taxYear,
+            date: $event->date,
+            capitalGain: new CapitalGain($event->costBasis, $event->proceeds),
+        ));
```

Comment:
ℹ️ Whether it's a gain or a loss now boils down to the difference between the cost basis and the proceeds being positive or negative. That difference is worked out by the `CapitalGain` instance directly